mlflow/utils/search_utils.py

```diff
@@ -1910,42 +1910,118 @@ def model_matches(model):
 
     @classmethod
-    def parse_order_by_for_logged_models(cls, order_by):
-        token_value, is_ascending = cls._parse_order_by_string(order_by)
-        token_value = token_value.strip()
-        if token_value not in cls.VALID_ORDER_BY_ATTRIBUTE_KEYS:
+    def parse_order_by_for_logged_models(
+        cls, order_by: dict[str, Any]
+    ) -> tuple[str, bool, str, str]:
+        if not isinstance(order_by, dict):
             raise MlflowException.invalid_parameter_value(
-                f"Invalid order by key '{token_value}' specified. Valid keys "
-                f"are '{cls.VALID_ORDER_BY_ATTRIBUTE_KEYS}'",
+                "`order_by` must be a list of dictionaries."
             )
-        identifier = cls._get_identifier(token_value, cls.VALID_ORDER_BY_ATTRIBUTE_KEYS)
-        return identifier["type"], identifier["key"], is_ascending
+        field_name = order_by.get("field_name")
+        if field_name is None:
+            raise MlflowException.invalid_parameter_value(
+                "`field_name` in the `order_by` clause must be specified."
+            )
+        if "." in field_name:
+            entity = field_name.split(".", 1)[0]
+            if entity != "metrics":
+                raise MlflowException.invalid_parameter_value(
+                    f"Invalid order by field name: {entity}, only `metrics.<name>` is allowed."
+                )
+        else:
+            field_name = field_name.strip()
+            if field_name not in cls.VALID_ORDER_BY_ATTRIBUTE_KEYS:
+                raise MlflowException.invalid_parameter_value(
+                    f"Invalid order by field name: {field_name}."
+                )
+        ascending = order_by.get("ascending", True)
+        if ascending not in [True, False]:
+            raise MlflowException.invalid_parameter_value(
+                "Value of `ascending` in the `order_by` clause must be a boolean, got "
+                f"{type(ascending)} for field {field_name}."
+            )
+        dataset_name = order_by.get("dataset_name")
+        dataset_digest = order_by.get("dataset_digest")
+        if dataset_digest and not dataset_name:
+            raise MlflowException.invalid_parameter_value(
+                "`dataset_digest` can only be specified if `dataset_name` is also specified."
+            )
+        return field_name, ascending, dataset_name, dataset_digest
+
+    @classmethod
+    def _apply_reversor_for_logged_model(
+        cls,
+        model: LoggedModel,
+        key: str,
+        ascending: bool,
+        dataset_name: Optional[str],
+        dataset_digest: Optional[str],
+    ):
+        if "." in key:
+            metric_key = key.split(".", 1)[1]
+            metric_values = [
+                m.value
+                for m in model.metrics
+                if m.key == metric_key
+                and (dataset_name is None or m.dataset_name == dataset_name)
+                and (dataset_digest is None or m.dataset_digest == dataset_digest)
```

Comment:
```suggestion
                and (not dataset_name or m.dataset_name == dataset_name)
                and (not dataset_digest or m.dataset_digest == dataset_digest)
```
@harupy Maybe we need this change to support empty string case?

Reply:
yea let's ignore an empty string.